src/osd/CMakeLists.txt

```diff
@@ -44,8 +44,9 @@ add_library(osd STATIC ${osd_srcs}
   $<TARGET_OBJECTS:cls_references_objs>
   $<TARGET_OBJECTS:global_common_objs>)
 target_link_libraries(osd PRIVATE
+  os dmclock heap_profiler cpu_profiler
   ${LEVELDB_LIBRARIES}
-  dmclock heap_profiler cpu_profiler ${CMAKE_DL_LIBS})
+  ${CMAKE_DL_LIBS})
```

Comment:
i see that we still link os in `target_link_libraries(ceph-osd osd os`, can we make this one PUBLIC so it isn't needed in ceph-osd?
```cmake
target_link_libraries(osd
  PUBLIC os
  PRIVATE dmclock heap_profiler cpu_profiler
```